internal/services/apicenter/api_center_environment_resource.go

```diff
@@ -0,0 +1,304 @@
+	DevPortalUri   string `tfschema:"development_portal_uri"`
+	Instructions   string `tfschema:"instructions"`
+	ServerType     string `tfschema:"server_type"`
+	MgmtPortalUri  string `tfschema:"management_portal_uri"`
+}
+
+func (r ApiCenterEnvironmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"service_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"identification": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"environment_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice(
+				environments.PossibleValuesForEnvironmentKind(),
+				false),
+		},
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"development_portal_uri": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPS,
+		},
+
+		"instructions": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"server_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice(
+				environments.PossibleValuesForEnvironmentServerType(),
+				false),
+		},
+		"management_portal_uri": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPS,
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ApiCenterEnvironmentResource) ModelObject() interface{} {
+	return &ApiCenterEnvironmentResourceModel{}
+}
+
+func (r ApiCenterEnvironmentResource) ResourceType() string {
+	return "azurerm_api_center_environment"
+}
+
+func (r ApiCenterEnvironmentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return environments.ValidateEnvironmentID
+}
+
+func (r ApiCenterEnvironmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			service, err := services.ParseServiceID(model.ServiceId)
+			if err != nil {
+				return err
+			}
+
+			// @favoretti: can't find any workspace creation buttons in the portal, assume everything defaults to "default" for now?
+			id := environments.NewEnvironmentID(subscriptionId, service.ResourceGroupName, service.ServiceName, "default", model.Identification)
```

Comment:
It doesn't make sense to me why we're exposing a `name` property but are using `identification` to build the ID

Reply:
There are 2 fields in the portal/API - name and identification. When you create an API Center environment they default to the same value, but if you change either name or identification - the identification value is the one that's being used in the resource ID, name can be an arbitrary thing.

Doesn't make a whole lot of sense to me either, but this is the way API works currently.

Reply:
There's also inconsistency between what people see in portal (Name/Identification) and field names in the API (Title/Identification)